### PR TITLE
PremiumClub Parsing

### PR DESCRIPTION
**Pull request by Zintixx**
- Parsing Premium Club packages and daily items.
- Removed Player.isVIP to check on a player's PremiumClub status and using the player's premiumclub expiration instead

### CODE REVIEW

File: MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs [hunk 1]
```diff
@@ -64,12 +84,49 @@ private static void HandleOpenPurchaseWindow(GameSession session)
...
+            {
+                Item bonusItem = new(item.Id)
+                {
+                    Rarity = item.Rarity,
+                    Amount = item.Amount
+                };
+                InventoryController.Add(session, bonusItem, true);
+            }
+
+            long expiration = DateTimeOffset.UtcNow.ToUnixTimeSeconds() + (vipPackage.VipPeriod * 3600); // Convert to seconds, as vipPeriod is given as hours
```
Comment (AlanMorel) [inline MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs:113]: You can make `vipPackage.VipPeriod * 3600` a variable then use here and on line 121

File: MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs [hunk 2]
```diff
@@ -64,12 +84,49 @@ private static void HandleOpenPurchaseWindow(GameSession session)
...
+
+            if (session.Player.VIPExpiration < DateTimeOffset.UtcNow.ToUnixTimeSeconds())
```
Comment (AlanMorel) [inline MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs:115]: Looks this could be replaced with:
`if (!IsVip(session.Player)) {`

**Review by AlanMorel — CHANGES_REQUESTED** [review]

**Review by AlanMorel — APPROVED** [review]

**Review by Sparkymod — APPROVED** [review]

File: MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs [hunk 3]
```diff
@@ -64,12 +84,50 @@ private static void HandleOpenPurchaseWindow(GameSession session)
...
+                session.Player.VIPExpiration += vipTime;
+            }
+
+            session.Send(PremiumClubPacket.ActivatePremium(session.FieldPlayer, session.Player.VIPExpiration));
+        }
+
+        public static bool IsVip(Player player)
+        {
+            return player.VIPExpiration > DateTimeOffset.UtcNow.ToUnixTimeSeconds();
         }
```
Comment (angelotadeucci) [inline MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs:131]: Shouldn't this function be in Player instead?

File: MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs [hunk 4]
```diff
@@ -64,12 +84,50 @@ private static void HandleOpenPurchaseWindow(GameSession session)
...
+            int vipId = packet.ReadInt();
+
+            if (!PremiumClubPackageMetadataStorage.IsValid(vipId))
+            {
+                return;
+            }
+
+            PremiumClubPackageMetadata vipPackage = PremiumClubPackageMetadataStorage.GetMetadata(vipId);
+
+            if (!session.Player.Wallet.Meret.Modify(vipPackage.Price))
```
Comment (angelotadeucci) [inline MapleServer2/PacketHandlers/Game/PremiumClubHandler.cs:96]: If it's possible to buy VIP with any type of Meret you need to use `Wallet.RemoveMerets()`

**Review by angelotadeucci — CHANGES_REQUESTED** [review]

File: MapleServer2/Types/Player.cs [hunk 5]
```diff
@@ -382,5 +382,10 @@ private PlayerStat AddStatRegen(PlayerStatId statIndex, PlayerStatId regenStatIn
             int postRegen = Math.Clamp(stat.Current + regen, 0, stat.Max);
             return new PlayerStat(stat.Max, stat.Min, postRegen);
         }
+
+        public static bool IsVip(Player player)
+        {
+            return player.VIPExpiration > DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+        }
```
Comment (angelotadeucci) [inline MapleServer2/Types/Player.cs:389]: I would prefer something like this.
```suggestion
        public bool IsVip()
        {
            return VIPExpiration > DateTimeOffset.UtcNow.ToUnixTimeSeconds();
        }
```
And then you change in PremiumClubHandler.cs to `if (!session.Player.IsVip())`

**Review by angelotadeucci — CHANGES_REQUESTED** [review]

**Review by angelotadeucci — APPROVED** [review]
👌

**Pull request merged**